library/core/src/ffi/c_str.rs

```diff
@@ -551,7 +550,8 @@ impl CStr {
     #[must_use = "this returns the result of the operation, \
                   without modifying the original"]
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_const_unstable(feature = "const_cstr_methods", issue = "101719")]
+    #[rustc_allow_const_fn_unstable(const_slice_index)]
```

Comment:
this indirectly uses `const_trait_impls`, we should not stabilize this.

Reply:
I'll remove this function from this PR.

Could you take another look at #107429 in that case? You had approved the changes, but it does also rely on this feature gate. If we need to revert just the constness part of that PR we can do that, it just merged yesterday

Reply:
In both these cases, I think we could keep them const by just replacing the indexing calls with pointer math. Or maybe it would mean this feature gate is OK since that is an option to keep const? 

Let me know what's best

Reply:
Yea, it could be replaced by pointer shenanigans, but doing so now makes the functions harder to read. I guess if it's ok with everyone during the FCP that we'd fall back to that in the worst case...

Reply:
As of 5cb701f3794b04fe35c7fcf1f6e2a0e2c5950415 I replaced the `const_slice_index` call with the pointer version